internal/services/users-service.go

```diff
@@ -56,6 +57,22 @@ func (s *UsersService) NewUser(input ports.CreateUserInput) *customErrors.Intern
 	return nil
 }
 
+func (s *UsersService) GetUser(id string) (*entities.User, *customErrors.InternalError) {
+	if !validators.IsValidUuid(id) {
+		return nil, customErrors.NewInternalError("invalid uuid", 400, []string{})
+	}
+
+	response, err := s.repository.FindById(id)
+
+	if err != nil {
+		return nil, customErrors.NewInternalError("a failure occurred when try to retrieve a new user", 500, []string{err.Error()})
+	} else if response == nil {
+		return nil, customErrors.NewInternalError("User not found", 404, []string{})
```

Comment:
Aparentemente o padrão da Golang com erros é letras minúsculas. Mesmo `User` sendo uma entidade, não há necessidade de usar o CamelCase.
```suggestion
		return nil, customErrors.NewInternalError("user not found", 404, []string{})
```